src/Core/Managed/Shared/Channel/TelemetryBuffer.cs

```diff
@@ -46,7 +48,30 @@ public int Capacity
                 this.capacity = value;
             }
         }
-        
+
+        /// <summary>
+        /// Gets or sets the maximum number of telemetry items that can be in the backlog to send. Items will be dropped
+        /// once this limit is hit.
+        /// </summary>        
+        public int MaximumUnsentBacklogSize
```

Comment:
Items already "Sent" after you pass them into `Send` method. `Unsent` in the name doesn't make much sense.